fixlib/fixlib/baseresources.py

```diff
@@ -719,6 +722,23 @@ def __getstate__(self) -> Dict[str, Any]:
     def __setstate__(self, state: Dict[str, Any]) -> None:
         self.__dict__.update(state)
 
+    @classmethod
+    def get_all_categories(cls: Type["BaseResource"]) -> List[Category]:
+        def gather_categories(class_type: Type["BaseResource"]) -> List[Category]:
+            merged = []
+            for base in class_type.__bases__:
+                if issubclass(base, BaseResource):
+                    merged.extend(gather_categories(base))
+            if hasattr(class_type, "_categories"):
+                merged.extend(class_type._categories)
+            return merged
+
+        return list(set(gather_categories(cls)))
+
+    @property
+    def categories(self) -> List[str]:
+        return [str(category.value) for category in self.get_all_categories()]
```

Comment:
```suggestion
        return [category.value.name for category in self.get_all_categories()]
```
To not rely on the str representation?

Reply:
I purposefully chose str() here (and anywhere else) because it allows me to change the (already existing) implementation of __str__ in one place and it changes throughout the code. If I use .name here and later decide I'd rather use .long_name or .description or something else as the string representation, I might miss it in one place and wouldn't necessarily see an error, just inconsistent data.